include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -972,8 +973,10 @@ SplineBuilder<
         IDimX...>::quadrature_coefficients() const
 {
     // Compute integrals of bsplines
-    ddc::Chunk integral_bsplines(spline_domain(), ddc::KokkosAllocator<double, MemorySpace>());
-    ddc::discrete_space<bsplines_type>().integrals(integral_bsplines.span_view());
+    ddc::Chunk integral_bsplines_host(spline_domain(), ddc::HostAllocator<double>());
+    ddc::discrete_space<bsplines_type>().integrals(integral_bsplines_host.span_view());
```

Comment:
I wonder if the function `integrals` actually needs to be `KOKKOS_FUNCTION` ?

Reply:
> I wonder if the function integrals actually needs to be KOKKOS_FUNCTION ?

It probably doesn't have to be as it will usually only be used during the initialisation but I agree that it would make sense to do this

Reply:
> Maybe a `static_assert` should also be added inside `integrals` to prevent it being called with a device chunkspan

It can be called with a device chunkspan, as long as the call is made from the device. Same for host, it can be called with a host chunkspan as long as the call is made from the host.

> Why would you need this to compile?

It would but then the `KOKKOS_INLINE_FUNCTION` annotation is useless. But maybe it would be a good thing.

Reply:
Ah ok. I see what you mean. As this is an inline function is there anyway to get the execution space so you can do assert that it is compatible with the memory space?

Reply:
It is possible indeed, Kokkos uses `KOKKOS_IF_ON_HOST`/`KOKKOS_IF_ON_DEVICE`  to do that in `Kokkos::View`.

Looking at our usage of this function:
- include/ddc/kernels/splines/spline_evaluator.hpp:384
- include/ddc/kernels/splines/spline_evaluator_2d.hpp:823
- include/ddc/kernels/splines/spline_builder.hpp:976

We see that we need more a host function that can deal with a ChunkSpan on any memory space but does not need to be callable from the device.

Reply:
If you are ok with this, I will open a new issue about this.

Reply:
Yes that's fine for me

Reply:
see https://github.com/CExA-project/ddc/issues/615#issue-2502666763